src/Imports/ModelManager.php

```diff
@@ -66,10 +66,15 @@ public function flush(ToModel $import, bool $massInsert = false)
      * @param ToModel $import
      * @param array   $attributes
      *
+     * @param int|null $rowNumber
      * @return Model[]|Collection
      */
-    public function toModels(ToModel $import, array $attributes): Collection
+    public function toModels(ToModel $import, array $attributes, $rowNumber = null): Collection
     {
+        if (method_exists($import, 'setRowNumber')) {
```

Comment:
Perhaps we can call this method once (where you had it before) and store the result in a class property `$this->remembersRowNumber`